### DISCUSSION TITLE
Tasmota32 with option USE_KEELOQ

### DISCUSSION THREAD

**Discussion by wjohn007** [arendst/Tasmota#21505] [Q&A]
I am trying to build Tasmota32 with Keeloq option using GitPot using Tasmota master branch with following extension in platformio_tasmota_cenv32.ini file.

```
	[env:tasmota32-keeloq]
	extends                 = env:tasmota32_base
	build_flags             = ${env:tasmota32_base.build_flags}
				-DFIRMWARE_TASMOTA32
				-DUSE_KEELOQ
				-DOTA_URL='"http://ota.tasmota.com/tasmota32/release/tasmota32.bin"'
```

The build works and creates the tasmota32-keeloq.bin file.
Unfortunately, the components "CC1101 GDO0" and "CC1101 GDO2" are not offered in the component list when I try to configure a GPIO.
![grafik](https://github.com/arendst/Tasmota/assets/115527085/d2a5a9be-a4de-4989-a008-9d0b81f41d55)

What's going wrong?

John

**Comment by barbudor** [comment]
Your problem is
```
				-DFIRMWARE_TASMOTA32
```
which will override your USE_KEELOQ

https://github.com/arendst/Tasmota/blob/e07b0e42b3f3ec0af76cc30f4627c02dcad614ba/tasmota/include/tasmota_configurations_ESP32.h#L565-L595

**Reply by barbudor** [reply]
You should use `user_config_override.h`  as recommended

**Reply by sfromis** [reply]
This line from `tasmota_globals.h` suggests that this combo is not supported.
```
#undef USE_KEELOQ                   // Disable support for Jarolift rollers by Keeloq algorithm as it's library cc1101 is not compatible with ESP32
```

**Reply by wjohn007** [reply]
Thank you for the helpful hints
